apps/wallet/src/shared/qredo-api.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+export type QredoAPIErrorResponse = {
+    code: string;
+    msg: string;
+    detail: {
+        reason: string;
+    };
+};
+
+export class QredoAPIError extends Error {
```

Comment:
just an FYI: you might have to do some weird stuff if you want to be able to do `instanceof` checks with these custom exceptions: https://stackoverflow.com/questions/31626231/custom-error-class-in-typescript

Reply:
I've heard this a lot but I haven't actually seen this be an issue in the latest browsers?

Reply:
It was done like this to make more detailed error, it's not used with instance of at the moment. Will leave it as is for now and revisit in case it creates any problems